drivers/spider.py

```diff
@@ -0,0 +1,114 @@
+"""
+  Spider3 Rider Driver
+"""
+
+import sys
+import os
+import inspect
+
+OSGAR_ROOT = os.path.realpath(os.path.abspath(os.path.join(os.path.split(inspect.getfile( inspect.currentframe() ))[0],"..")))
+if OSGAR_ROOT not in sys.path:
+    sys.path.insert(0, OSGAR_ROOT) # access to logger without installation
+
+
+import serial
+import struct
+from threading import Thread, Event
+
+from lib.logger import LogWriter, LogReader
+
+
+class Spider(Thread):
+    def __init__(self, config, logger, output, name='spider'):
+        Thread.__init__(self)
+        self.setDaemon(True)
+        self.should_run = Event()
+        self.should_run.set()
+
+        self.logger = logger
+        self.stream_id_in = config['stream_id_in']
+        self.stream_id_out = config['stream_id_out']
+
+        self.buf = b''
+        self.output = output
+        self.name = name
+
+        if 'port' in config:
+            self.com = serial.Serial(config['port'], config['speed'], dsrdtr=1)
+            self.com.setRTS()
+            self.com.setDTR(0)
+            self.com.timeout = 0.01  # expects updates < 100Hz
+
+            for i in range(10):
+                data = self.com.read(1024)
+                self.logger.write(self.stream_id_in, data)
```

Comment:
Why is this for cycle here?

Reply:
Well, I could not make it to work - the CAN bridge reset is triggered by `self.com.setDTR(0)` and I had no idea how long it will take to be ready to accept commands. This code helped me to get this log:
```
python lib\logger.py d:\md\osgar\logs\jk\ro2018-180202_150001.log --times --stream 1
0:00:00.013960 1 b''
0:00:00.024248 1 b''
0:00:00.034550 1 b''
0:00:00.044857 1 b''
0:00:00.055141 1 b''
0:00:00.065428 1 b'\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xfe\x10'
0:00:00.075718 1 b''
0:00:00.086056 1 b''
0:00:00.096333 1 b''
0:00:00.106609 1 b''
```
I am not sure whether to add it in `run()`? After 0xFE 0x10 it is necessary to send 10 times 0xFF and configure CAN communication speed.

Reply:
Can we embed this info into some documentation? It is not obvious from the code itself.